### PR TITLE
Fix SLURM RAM monitoring

### LINKED ISSUE #447: CodeCarbon spams Slurm controler with `scontrol`
Hello,

We are facing some issues with CodeCarbon in our facility because it spams our Slurm controler with `scontrol`.

If rate-limiting is enabled this also means that the jobs that use CodeCarbon are slowed down because they spend a lot of time waiting for `scontrol` to return.

As far as I could tell, the values that are read from `scontrol` won't change during the run so they could be cached by CodeCarbon instead of being fetched everytime.

We are willing to provide a PR if needed. If caching is used somewhere else, feel free to point us to the relevant code so that we use a similar implementation.

Rémi

### PR DESCRIPTION
**Pull request by benoit-cty**
Fix SLURM RAM monitoring :
- Fix error when calling scontrol
- Do only one call instead of one at each measure

Will close #447

### CODE REVIEW

**Comment by benoit-cty** [comment]
Tested on a SLURM environment : there is only too call per sessions and no more error message.

File: codecarbon/external/test.ipynb
Comment (SaboniAmine) [inline codecarbon/external/test.ipynb:1]: Can you clean this file ? Or move it to test folder ?

**Review by SaboniAmine — COMMENTED** [review]
Thanks Benoît! Small comment about the notebook, not sure we want to merge it to master

File: codecarbon/external/test.ipynb
Comment (benoit-cty) [reply codecarbon/external/test.ipynb:1]: Thanks, I will delete it.

**Comment by vict0rsch** [comment]
I'll have a look next week 😊

**Comment by vict0rsch** [comment]
There was an issue with `$SLURM_JOBID` which was changed to `$SLURM_JOB_ID`, making some queries like `scontrol show job $SLURM_JOBID"` fail.

**Comment by vict0rsch** [comment]
This works as expected

```
from codecarbon.external.hardware import RAM

ram = RAM(tracking_mode="process")
ram.slurm_memory_GB
```

**Comment by benoit-cty** [comment]
> There was an issue with `$SLURM_JOBID` which was changed to `$SLURM_JOB_ID`, making some queries like `scontrol show job $SLURM_JOBID"` fail.

Thanks for your review, it seems to be deprecated since 2017 so it is safe to remove the support of the old name !

File: codecarbon/external/test.ipynb
Comment (benoit-cty) [reply codecarbon/external/test.ipynb:1]: It's deleted. Can you approve the PR if everything is OK ?
It has been rebased.

**Review by vict0rsch — APPROVED** [review]

**Pull request merged**